Warnings on clang 3.7

Hi,

Just compiled core with clang 3.7, and I noticed a few new warnings regarding some `move` statements:

```
In file included from column.cpp:12:
In file included from ../realm/query_engine.hpp:98:
../realm/table_view.hpp:370:16: warning: moving a local object in a return statement prevents copy elision [-Wpessimizing-move]
        return move(retval);
               ^
../realm/table_view.hpp:370:16: note: remove std::move call here
        return move(retval);
               ^~~~~      ~
../realm/table_view.hpp:378:16: warning: moving a local object in a return statement prevents copy elision [-Wpessimizing-move]
        return move(retval);
               ^
../realm/table_view.hpp:378:16: note: remove std::move call here
        return move(retval);
               ^~~~~      ~
../realm/table_view.hpp:637:16: warning: moving a local object in a return statement prevents copy elision [-Wpessimizing-move]
        return move(retval);
               ^
../realm/table_view.hpp:637:16: note: remove std::move call here
        return move(retval);
               ^~~~~      ~
../realm/table_view.hpp:645:16: warning: moving a local object in a return statement prevents copy elision [-Wpessimizing-move]
        return move(retval);
               ^
../realm/table_view.hpp:645:16: note: remove std::move call here
        return move(retval);
               ^~~~~      ~
```

--- a/src/realm/table_view.hpp
+++ b/src/realm/table_view.hpp
@@ -367,7 +367,7 @@ protected:
     {
         patch.reset(new Handover_patch);
         std::unique_ptr<TableViewBase> retval(new TableViewBase(*this, *patch, mode));
-        return move(retval);
+        return retval;
     }
 
     virtual std::unique_ptr<TableViewBase> clone_for_handover(std::unique_ptr<Handover_patch>& patch,
@@ -375,7 +375,7 @@ protected:
     {
         patch.reset(new Handover_patch);
         std::unique_ptr<TableViewBase> retval(new TableViewBase(*this, *patch, mode));
-        return move(retval);
+        return retval;
     }
 
     virtual void apply_and_consume_patch(std::unique_ptr<Handover_patch>& patch, Group& group)
@@ -634,7 +634,7 @@ public:
     {
         patch.reset(new Handover_patch);
         std::unique_ptr<TableViewBase> retval(new ConstTableView(*this, *patch, mode));
-        return move(retval);
+        return retval;
     }
 
     std::unique_ptr<TableViewBase> 
@@ -642,7 +642,7 @@ public:
     {
         patch.reset(new Handover_patch);
         std::unique_ptr<TableViewBase> retval(new ConstTableView(*this, *patch, mode));
-        return move(retval);
+        return retval;
     }
 
     // this one is here to follow the general scheme, it is not really needed, the

